packages/server-core/src/networking/instance-provision/instance-provision.class.ts

```diff
@@ -202,6 +204,46 @@ export async function checkForDuplicatedAssignments(
     }
   }
 
+  // This is here to handle odd cases with externally unresponsive instanceserver pods.
+  // It tries to make a GET request to the pod. If there's an error, or the response takes more than 2 seconds,
+  // it assumes the pod is unresponsive. Locally, it just waits half a second and tries again - if the local
+  // instanceservers are rebooting after the last person left, we just need to wait a bit for them to start.
+  // In production, it attempts to delete that pod via the K8s API client and tries again.
+  let responsivenessCheck: boolean
+  responsivenessCheck = await Promise.race([
+    new Promise<boolean>((resolve) => {
+      setTimeout(() => {
+        logger.warn(`Instanceserver at ${ipAddress} too long to respond, assuming it is unresponsive and killing`)
+        resolve(false)
+      }, 2 * 1000) // timeout after 2 seconds
```

Comment:
do we want to make these times configurable?

Reply:
Made it so.